osu.Framework/RuntimeInfo.cs

```
@@ -28,39 +26,17 @@ public static class RuntimeInfo
         static RuntimeInfo()
         {
             IsMono = Type.GetType("Mono.Runtime") != null;
-            int p = (int)Environment.OSVersion.Platform;
-            IsUnix = p == 4 || p == 6 || p == 128;
-            IsWindows = Path.DirectorySeparatorChar == '\\';
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                IsWindows = true;

```

Why not `IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);` and so on, and then throw an exception if more than 1 of the 3 booleans are `true`?

---

Sure, that's another path you can take with this.

---

As we start to not trust libraries we use to do the right thing. Gotta draw some lines on what we trust and what we don't, no?

---

If the issue thread you linked fails, it will still fail. There is no line drawn here.

---

The reason I am mentioning this is that with the current impl multiple true cases would be treated at just one, which may silently fail later on. Of course it's unlikely that `RuntimeInformation.IsOSPlatform` gives weird results like that, but it's just in general a good idea to sanitize input and gracefully handle edge cases.

---

My argument was that these methods may be doing potentially expensive queries (ie. doing something similar what we were doing, which involved executing an external process), so we should optimise to run the minimum number required.

---

Fair point. I did not consider that. I think that overrules my argument by being more important for the user experience.

---

Let's just not use any libraries then because we don't know what any library does internally. I think there is a minimum level of trust you have to give to a library before you performance profile to determine that it actually is doing an expensive query.

---

And if we're being honest with ourselves, "executing an external process" is not an expensive task.